File: android/src/main/kotlin/com/pravera/flutter_bpass_lib/utils/ErrorHandleUtils.kt
package com.pravera.flutter_bpass_lib.utils

import com.pravera.flutter_bpass_lib.errors.ErrorCodes
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodChannel

class ErrorHandleUtils {
	companion object {
		fun handleMethodCallError(result: MethodChannel.Result?, errorCode: ErrorCodes) {
			result?.error(errorCode.toString(), errorCode.message(), null)
		}

		fun handleStreamError(events: EventChannel.EventSink?, errorCode: ErrorCodes) {
			events?.error(errorCode.toString(), errorCode.message(), null)
		}
	}
}

File: ios/Classes/utils/ErrorHandleUtils.swift
//
//  ErrorHandleUtils.swift
//  flutter_bpass_lib
//
//  Created by WOO JIN HWANG on 2022/05/04.
//

import Flutter
import Foundation

class ErrorHandleUtils {
  static func handleMethodCallError(result: FlutterResult, errorCode: ErrorCodes) {
    let error = FlutterError(code: errorCode.rawValue, message: errorCode.message(), details: nil)
    result(error)
  }
  
  static func handleStreamError(events: FlutterEventSink, errorCode: ErrorCodes) {
    let error = FlutterError(code: errorCode.rawValue, message: errorCode.message(), details: nil)
    events(error)
  }
}
